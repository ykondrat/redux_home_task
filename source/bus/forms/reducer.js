// Core
import { combineForms } from 'react-redux-form';

export const formsReducer = combineForms({
    filter: {
        message: ''
    },
    updateTask: {
        message: ''
    }
}, 'forms');
