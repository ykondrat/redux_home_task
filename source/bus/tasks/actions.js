// Types
import { types } from './types';

// Async
export const fetchTasksAsync = () => ({
    type: types.FETCH_TASKS_ASYNC
});
export const createTaskAsync = (message) => ({
    type: types.CREATE_TASK_ASYNC,
    payload: message
});
export const removeTaskAsync = (taskId) => ({
    type: types.REMOVE_TASK_ASYNC,
    payload: taskId
});
export const updateTaskAsync = (task) => ({
    type: types.UPDATE_TASK_ASYNC,
    payload: task
});
export const completeTasksAsync = (tasks) => ({
    type: types.COMPLETE_TASKS_ASYNC,
    payload: tasks
});

// Sync
export const fillTasks = (tasks) => ({
    type: types.FILL_TASKS,
    payload: tasks
});
export const createTask = (task) => ({
    type: types.CREATE_TASK,
    payload: task
});
export const removeTask = (taskId) => ({
    type: types.REMOVE_TASK,
    payload: taskId
});
export const updateTask = (task) => ({
    type: types.UPDATE_TASK,
    payload: task
});
export const completeTask = () => ({
    type: types.COMPLETE_TASKS,
});
