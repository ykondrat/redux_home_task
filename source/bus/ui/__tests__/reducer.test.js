// Reducer
import { uiReducer } from '../reducer';

import * as actions from '../actions';

const actionTest = {
    type: 'INIT_TEST_DEFAULT'
}

describe('uiReducer tests:', () => {
    test('uiReducer test for default switch case', () => {
        expect(
            uiReducer( void 0, actionTest)
        ).toMatchSnapshot();
    });

    test('uiReducer test for startFetching', () => {
        expect(
            uiReducer(void 0, actions.startFetching())
        ).toMatchSnapshot();
    });

    test('uiReducer test for stopFetching', () => {
        expect(
            uiReducer(void 0, actions.stopFetching())
        ).toMatchSnapshot();
    });

});
