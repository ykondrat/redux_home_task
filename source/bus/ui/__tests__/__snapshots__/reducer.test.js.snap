// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uiReducer tests: uiReducer test for default switch case 1`] = `
Immutable.Map {
  "isFetching": false,
}
`;

exports[`uiReducer tests: uiReducer test for startFetching 1`] = `
Immutable.Map {
  "isFetching": true,
}
`;

exports[`uiReducer tests: uiReducer test for stopFetching 1`] = `
Immutable.Map {
  "isFetching": false,
}
`;
